src/test/java/com/desiato/puresynth/models/UserControllerTest.java

```diff
@@ -0,0 +1,57 @@
+package com.desiato.puresynth.models;
+
+import com.desiato.puresynth.configurations.SecurityConfig;
+import com.desiato.puresynth.services.UserService;
+import org.junit.jupiter.api.Test;
+import com.desiato.puresynth.controllers.UserController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Import;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@WebMvcTest(controllers = UserController.class)
+@AutoConfigureMockMvc
+@Import(SecurityConfig.class)
+public class UserControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private UserService userService;
+
+    @Test
+    public void createUser_ShouldReturnCreatedUser() throws Exception {
+        String newUserJson =
```

Comment:
This request body should be closer to `mockMvc.perform(post("/api/user")`. Code executed and related to each other should be placed together.